tests/test_add.py

```diff
@@ -402,3 +402,31 @@ def is_symlink_true_below_dvc_root(path):
 
             stage_file_path = self.DATA + Stage.STAGE_FILE_SUFFIX
             self.assertTrue(os.path.exists(stage_file_path))
+
+
+class TestAddFilename(TestDvc):
+    def test(self):
+        ret = main(["add", self.FOO, self.BAR, "-f", "error.dvc"])
+        self.assertNotEqual(0, ret)
+
+        ret = main(["add", "-R", self.DATA_DIR, "-f", "error.dvc"])
+        self.assertNotEqual(0, ret)
+
+        with self.assertRaises(DvcException):
```

Comment:
Let's create a new exception for that error and catch that one, instead of catching generic DvcException, since it can be raised in any other place.